crates/polars-ops/src/chunked_array/array/to_struct.rs

```diff
@@ -0,0 +1,41 @@
+use polars_core::export::rayon::prelude::*;
+use polars_utils::format_smartstring;
+use smartstring::alias::String as SmartString;
+
+use super::*;
+
+pub type ArrToStructNameGenerator = Arc<dyn Fn(usize) -> SmartString + Send + Sync>;
+
+pub fn arr_default_struct_name_gen(idx: usize) -> SmartString {
+    format_smartstring!("field_{idx}")
+}
+
+pub trait ToStruct: AsArray {
+    fn to_struct(
+        &self,
+        name_generator: Option<ArrToStructNameGenerator>,
+    ) -> PolarsResult<StructChunked> {
+        let ca = self.as_array();
+        let n_fields = ca.width();
+
+        let name_generator = name_generator
+            .as_deref()
+            .unwrap_or(&arr_default_struct_name_gen);
+
+        polars_ensure!(n_fields != 0, ComputeError: "cannot create a struct with 0 fields");
+        let fields = (0..n_fields)
```

Comment:
Can you wrap this with `POOL.install()`?. We only want to run on the polars thread pool;.